apps/dsb-message-broker/src/addressbook/addressbook.service.ts

```diff
@@ -0,0 +1,49 @@
+import { ModuleRef } from '@nestjs/core';
+import { Injectable, OnModuleInit } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { ITransport, Channel } from '@energyweb/dsb-transport-core';
+import { NatsJetstreamAddressBook } from '@energyweb/dsb-address-book-nats-js';
+
+@Injectable()
+export class AddressBookService implements OnModuleInit {
```

Comment:
This service does not do anything, just wraps address book service, I suggest to follow this pattern https://github.com/energywebfoundation/dsb/blob/e267710c1f4bedd138412edbeb3ea873bda11363/apps/dsb-message-broker/src/transport/transport.module.ts#L31 so use providers to provide initialised AddressBook service.

Reply:
as we discussed because of dependency injection problem we cannot go with this approach.